keps/0034-instance-health.md

```diff
@@ -43,3 +43,58 @@ Drift detection (detecting that resource was deleted or changed manually)
 Including other types of resources than *readiness phase 1 resources*
 Determining if the underlying application is functional
 Determining if the underlying application is reachable
+
+## Proposal
+
+Readiness of an `Instance` will be communicated via a newly introduced `Status.Conditions` field. The schema of this field will [conform the schema](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1367) recommended by k8s api machinery. The `Type` of the newly added condition will be `Ready`. Condition is supposed to have these three values which will have the following meaning:
+True - the last observation of readiness on this instance is that it’s ready
+False - the last observation of readiness on this instance is that it’s NOT ready
+Unknown - deploy/upgrade/update plan is running
+
+Ready is expected to be an oscillating condition and it will indicate the object was believed to be fully operational at the time it was last verified. There’s an expectation that the value won’t be stalled for more than 1 minute (at least one minute after the state changes, status should be reflecting it).
+
+Unknown state will be set whenever deploy/upgrade/update plan is running - this is because plan run should be an atomic operation going from stable state to another stable state, evaluating health of all resources involved in the plan as part of the execution. It feels redundant to be also checking for readiness on a side. 
+
+The reason for using Conditions field rather than introducing a separate field is mainly that it’s starting to be established as a go-to pattern in kubernetes world. Also conditions have a standard set of fields attached with metadata that are useful for our case (namely Reason, Message). Having a ‘Ready’ condition is also [encouraged by sig-architecture](https://github.com/kubernetes/community/pull/4521/files). (excerpt from that document: *Although they are not a consistent standard, the `Ready` and `Succeeded` condition types may be used by API designers for long-running and bounded-execution objects, respectively.*)
+
+Example of Instance status being ready:
+```
+apiVersion: kudo.dev/v1beta1
+kind: Instance
+Status:
+  Conditions:
+    - type: Ready                     
+      status: "True"
+      lastTransitionTime: 2018-01-01T00:00:00Z
+```
+
+Example of Instance status being NOT ready:
+```
+apiVersion: kudo.dev/v1beta1
+kind: Instance
+Status:
+  Conditions:
+    - type: Ready                     
+      status: "False"
+      lastTransitionTime: 2018-01-01T00:00:00Z
+      Reason: ResourceNotReady
+      Message: Deployment ‘xxx’ is not ready, Deployment ‘yyy’ is not ready
+```
+
+### Implementation Details/Notes/Constraints
```

Comment:
I'm missing an explicit list of "readiness phase 1 resources" here somewhere

Reply:
should I move it here from the summary then? I wanted to have it only in one place 🤔

Reply:
Ah, it is there in the summary - never mind then, I was looking for an itemized list. I think here makes more sense, but i don't care too much as long as we have the list :)